internal/jpcommon/map.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jpcommon
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/jaegertracing/jaeger/proto-gen/storage/v2"
+)
+
+func PcommonMapToPlainMap(attributes pcommon.Map) map[string]string {
+	mapAttributes := make(map[string]string)
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		mapAttributes[k] = v.AsString()
+		return true
+	})
+	return mapAttributes
+}
+
+func PlainMapToPcommonMap(attributesMap map[string]string) pcommon.Map {
+	attributes := pcommon.NewMap()
+	for k, v := range attributesMap {
+		attributes.PutStr(k, v)
+	}
+	return attributes
+}
+
+// ConvertMapToKeyValues converts a pcommon.Map to []*storage.KeyValue.
+func ConvertMapToKeyValueList(m pcommon.Map) []*storage.KeyValue {
```

Comment:
similar to the comments in the reader, this package is turning into a kitchen sink with all kinds of dependencies. Let's keep a clean separation - only storage needs ConvertMapToKeyValueList, so define that helper there.